server/pages/index.js

```diff
@@ -370,11 +371,18 @@ module.exports = function() {
 			}
 		} );
 
-	app.get( '/browsehappy', function( request, response ) {
-		response.status( 200 ).render( 'browsehappy.jade', {
-			urls: generateStaticUrls( request )
+	app.get( '/browsehappy', setUpRoute, function( req, res, next ) {
+		const wpcomRe = /^https?:\/\/[A-z_-]+\.wordpress\.com$/;
+		const primaryBlogUrl = get( req, 'context.user.primary_blog_url', '' );
+		const isWpcom = wpcomRe.test( primaryBlogUrl );
+		req.context = Object.assign( {}, req.context, {
+			template: 'browsehappy.jade',
+			dashboardUrl: isWpcom
+				? primaryBlogUrl + '/wp-admin'
+				: 'https://dashboard.wordpress.com/wp-admin/'
 		} );
-	} );
+		next();
+	}, serverRender );
```

Comment:
While I agree having `serverRender` to be the last middleware in the chain everywhere, it's concerned with such different tasks that I'd rather use an inlined mw here that really only does the `res.render()`ing (much like e.g. [`render404`](https://github.com/Automattic/wp-calypso/blob/271a0e5a5badece497083b7801262e4989a08110/server/pages/index.js#L286-L290) does already).